src/game/client/camerashakesystem.cpp

```diff
@@ -0,0 +1,145 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief Class for managing camera shaking.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "camerashakesystem.h"
+#include <algorithm>
+
+#ifndef GAME_DLL
+CameraShakeSystemClass g_theCameraShakerSystem;
+#endif
+
+Vector3 s_axis_rotation(DEG_TO_RADF(7.5f), DEG_TO_RADF(15.0f), DEG_TO_RADF(5.0f));
+
+CameraShakeSystemClass::CameraShakerClass::CameraShakerClass(
+    Vector3 *position, float radius, float duration, float intensity) :
+    m_position(*position), m_radius(radius), m_duration(duration), m_intensity(intensity), m_time(0)
+{
+    m_vector1.X = GameMath::Random_Float(78.539818f, 94.24778f);
+    m_vector1.Y = GameMath::Random_Float(78.539818f, 94.24778f);
+    m_vector1.Z = GameMath::Random_Float(78.539818f, 94.24778f);
+
+    m_vector2.X = GameMath::Random_Float(0.0f, GAMEMATH_PI2);
+    m_vector2.Y = GameMath::Random_Float(0.0f, GAMEMATH_PI2);
+    m_vector2.Z = GameMath::Random_Float(0.0f, GAMEMATH_PI2);
+}
+
+void CameraShakeSystemClass::CameraShakerClass::Compute_Rotations(Vector3 *position, Vector3 *set_angles)
+{
+    captainslog_assert(set_angles != nullptr);
+
+    Vector3 pos_delta = *position - m_position;
+
+    float len2 = pos_delta.Length2();
+
+    if (m_radius * m_radius >= len2) {
+
+        float f1 = (1.0f - GameMath::Sqrt(len2) / m_radius) * m_intensity * (1.0f - m_time / m_duration);
+
+        for (int i = 0; i < 3; ++i) {
+
+            float f2 = (GAMEMATH_PI2 - m_vector1[i]) * m_time + m_vector1[i];
+            float f3 = f1 * s_axis_rotation[i];
+            float f4 = (f2 * m_time) + m_vector2[i];
+
+            (*set_angles)[i] += GameMath::Sin(f4) * f3;
+
+            Vector3 angles;
+
+            float f5 = f1 * 0.5f;
+
+            angles.X = GameMath::Random_Float(-f5, f5);
+            angles.Y = GameMath::Random_Float(-f5, f5);
+            angles.Z = GameMath::Random_Float(-f5, f5);
+
+            *set_angles += angles;
+        }
+    }
+}
+
+CameraShakeSystemClass::~CameraShakeSystemClass()
+{
+    while (!m_shakers.Is_Empty()) {
+        CameraShakerClass *shaker = m_shakers.Remove_Head();
+        m_shakers.Remove(shaker);
+        delete shaker;
+    }
+}
+
+bool CameraShakeSystemClass::Add_Camera_Shake(Vector3 *position, float radius, float duration, float amplitude)
+{
+    float intensity = amplitude * 3.1415927f / 180.0f;
```

Comment:
`DEG_TO_RADF(1.0f)` if I am not mistaken

Reply:
DEG_TO_RADF(amplitude)